src/software/jetson_nano/display/screens/home_screen.py

```diff
@@ -0,0 +1,169 @@
+from screen import Screen
+
+ROBOT_ID = 0
+CHANNEL_ID = 1
+
+PADDING = 6
+BASE_Y = 20
+BASE_X = 0
+BATTERY_VOLTAGE_BASE = 60
+
+"""
+This is the dashboard screen which shows basic robot diagnostics
+"""
+
+
+class HomeScreen(Screen):
+    def __init__(self, lcd_display, redis_dict, status_codes):
+        self.ids = [
+            redis_dict["robot id"],
+            redis_dict["channel id"],
+        ]
+        self.battery_voltage = redis_dict["battery voltage"]
+        self.cap_voltage = redis_dict["cap voltage"]
+        self.packet_loss = redis_dict["packet loss"]
+
+        def menu():
+            """ Action to go to Menu Screen """
+            return {self.status_codes["change screen"]: "Menu"}
+
+        def robot_id():
+            """ Set Robot ID """
+            return {
+                self.status_codes["edit"]: {
+                    "param": self.ids,
+                    "setting": ROBOT_ID,
+                    "delta": 1,
+                    "redis key": "robot id",
+                }
+            }
+
+        def channel_id():
+            """ Set Channel ID """
+            return {
+                self.status_codes["edit"]: {
+                    "param": self.ids,
+                    "setting": CHANNEL_ID,
+                    "delta": 1,
+                    "redis key": "channel id",
+                }
+            }
+
+        # Listing actions for Home Screen
+        self.actions = ["Robot ID", "Channel ID", "Menu"]
+        self.action_map = {
+            self.actions[0]: robot_id,
+            self.actions[1]: channel_id,
+            self.actions[2]: menu,
+        }
+
+        def draw_screen():
+            """ Home Screen Layout """
+            self.lcd_display.prepare()
+
+            cursor_size = self.font.getsize(">")[0]
+
+            x = cursor_size
+            y = 20
+
+            robot_id_str = "Robot ID"
```

Comment:
could we refactor this code a bit like set up robot id, set up battery voltage, set up capacitor voltage, etc?

Reply:
done